### PR TITLE
added linear search in C++

### PR DESCRIPTION
**Pull request by sidkhuntia**
I have added the code for linear search in C++.

### CODE REVIEW

File: c++/binary_search_sidkhuntia.cpp [hunk 1]
```diff
@@ -0,0 +1,44 @@
+#include <iostream> 
+
+#define lli long long int
+#define li long int
+#define ld long double
+#define f(i, n) for (int i; i < n; i++)
+
+using namespace std;
+const lli mod = 1e9 + 7;
```
Comment (king-11) [inline c++/binary_search_sidkhuntia.cpp:9]: This also has no correlation to searching remove this

File: c++/binary_search_sidkhuntia.cpp [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    }
+ 
+    return -1;
+}
+
+
+int main()
+{
+    ios_base::sync_with_stdio(0);
+    cin.tie(NULL);
```
Comment (king-11) [inline c++/binary_search_sidkhuntia.cpp:34]: Again not a contest so can remove these too

File: c++/binary_search_sidkhuntia.cpp [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+#define lli long long int
+#define li long int
+#define ld long double
+#define f(i, n) for (int i; i < n; i++)
+
+using namespace std;
+const lli mod = 1e9 + 7;
+
+int binarySearch(int arr[], int l, int r, int x)
```
Comment (king-11) [inline c++/binary_search_sidkhuntia.cpp:11]: I would prefer having function declaration above main while the function declaration goes below main as it makes sense to have main above all

File: c++/binary_search_sidkhuntia.cpp [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+
+int main()
+{
+    ios_base::sync_with_stdio(0);
+    cin.tie(NULL);
+    int arr[] = {1, 2, 3, 4, 5};
+    int x = 3;
+    int n = sizeof(arr) / sizeof(arr[0]);
```
Comment (king-11) [inline c++/binary_search_sidkhuntia.cpp:37]: Rather prefer taking a random `arr` and `n` as input

File: c++/binary_search_sidkhuntia.cpp [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+    int result = binarySearch(arr,0, n-1, x);
+    (result == -1)
+        ? cout << "Not available"
```
Comment (king-11) [inline c++/binary_search_sidkhuntia.cpp:41]: `Not available` sounds not so correct `Not Found` may sound better

File: c++/linear_search_sidkhuntia.cpp [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+using namespace std;
+const lli mod = 1e9 + 7;
+
+int search(int arr[], int n, int x)
+{
+    int i;
+    for (i = 0; i < n; i++)
+        if (arr[i] == x)
+            return i;
```
Comment (king-11) [inline c++/linear_search_sidkhuntia.cpp:16]: We can keep `i` as a local declaration

File: c++/linear_search_sidkhuntia.cpp [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+    return -1;
+}
+
+int main()
+{
+    ios_base::sync_with_stdio(0);
+    cin.tie(NULL);
+    int arr[] = {1, 2, 3, 4, 5};
+    int x = 3;
+    int n = sizeof(arr) / sizeof(arr[0]);
```
Comment (king-11) [inline c++/linear_search_sidkhuntia.cpp:26]: Same as binary search

File: c++/linear_search_sidkhuntia.cpp [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+
+    int result = search(arr, n, x);
+    (result == -1)
+        ? cout << "Not available"
```
Comment (king-11) [inline c++/linear_search_sidkhuntia.cpp:30]: Rewording

**Review by king-11 — CHANGES_REQUESTED** [review]

**Comment by king-11** [comment]
Moreover please include only one algorithm in your PR :)

File: c++/binary_search_sidkhuntia.cpp [hunk 9]
```diff
@@ -0,0 +1,44 @@
+#include <iostream> 
+
+#define lli long long int
+#define li long int
+#define ld long double
+#define f(i, n) for (int i; i < n; i++)
```
[the remark this answers is not in the document]
Comment (sidkhuntia) [reply c++/binary_search_sidkhuntia.cpp]: Can you please review now ?

**Comment by king-11** [comment]
@sidkhuntia updates on this you will have to make a new PR as the branch hasn't to non `main` and also ensure that algo is either binary or linear search only

**Pull request closed without merging**